### PR TITLE
Move common Psalm types

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

Related discussion - https://github.com/yiisoft/rbac-cycle-db/pull/60#discussion_r1287181997.

### CODE REVIEW

**Review by vjik — CHANGES_REQUESTED** [review]
Add new types to methods. For example, `ItemsStorageInterface::getParents()`

**Comment by arogachev** [comment]
@vjik Done.

File: src/AssignmentsStorageInterface.php [hunk 1]
```diff
@@ -7,6 +7,12 @@
 /**
  * `AssignmentsStorageInterface` represents a storage for assignment of RBAC items (a role or a permission) to a user
  * used in {@see Manager}.
+ *
+ * @psalm-type RawAssignment = array{
```
Comment (vjik) [inline src/AssignmentsStorageInterface.php:11]: Seems, it DB-specifc type. In other implementation type can be other. Right?

Comment (arogachev) [reply src/AssignmentsStorageInterface.php:11]: Yes. Removed `RawAssignment` and `RawItem`.

**Review by vjik — APPROVED** [review]

**Pull request merged**